### ISSUE TITLE
Custom field for TAG objects

### ISSUE THREAD

**Issue by fprina** [netbox-community/netbox#16080]
### NetBox version

v3.7.8

### Feature type

New functionality

### Proposed functionality

Permits extending tag information with custom fields, as is done with other object types.

### Use case

I use tags in Ansible playbook/templates and it would be great to add extra information based on them.

**Comment by jeffgdotorg** [comment]
Thank you for your interest in improving NetBox. We need more detail before we can consider implementing this request.

Please edit the issue body to expand your use case with Ansible usage examples showing how the proposed functionality would be used. The YAML example from the [`netbox.netbox.netbox_tag` module docs](https://docs.ansible.com/ansible/latest/collections/netbox/netbox/netbox_tag_module.html#ansible-collections-netbox-netbox-netbox-tag-module) might make a good starting point, but examples adapted from your own plays are also great.

**Comment by jeffgdotorg** [comment]
@fprina the revisions are a good start and I thank you for making them, but I need you to make a more concrete case. You need to sell a developer on the idea of spending an hour or two doing this work. Perhaps use a sanitized version of one of your existing plays as a starting point, and show how you imagine this enhancement would improve things and make your life (and other users' lives!) easier.

**Comment by fprina** [comment]
@jeffgdotorg, I understand your point, but in this case, I don't really have concrete details able to sell the idea.

My playbook is quite basic, just a lookup like 
`{{ query('netbox.netbox.nb_lookup', 'tags', api_endpoint=netbox_url, token=netbox_token, api_filter='color=00bcd4' ) }}`

but it would be better to use a clearer filter. 

For example, a boolean like `cf_firewall_group=true` would be great. 
Or maybe I could extend my implementation adding a text field with the firewall group name.

The recap is: looks like the only object type without CF are tag so would be nice have them. nothing more nothing less.

**Comment by jeffgdotorg** [comment]
The completeness argument suffices to move your issue along to `needs owner` status. If you would like to volunteer to work it through to a PR, please indicate and a maintainer will assign the issue to you. Otherwise, another developer with the requisite skills and capacity can pick it up.

**Comment by jeremystretch** [comment]
This would effect an API change for the Tag model, and therefore needs a milestone designation.

**Comment by goteamkor** [comment]
This would be a great addition. The ability to add custom fields to tags might also address the need for the following plugin : https://plugin-ideas.netbox.dev/ideas/PLUGINS-I-31

Rather than needing to create the plugin at all, or add a bunch of new models, you could simply 'tag' a device with a "contract" then add what ever custom fields for that tag that are appropriate for your environment/use case.

**Comment by antoinekh** [comment]
Could be very useful.
For example, we use tag to identify items to project code. with custom fields on tag we will be able to add more information on this tag

**Comment by PieterL75** [comment]
I once proposed to allow 'tags' on 'tags', just for this purpose.
then you can group all 'tags' that have one purpose with an extra tag on the tag (pfiew)
But having custom fields on tags would also solve this. 
+1 from me

**Comment by pchiquit** [comment]
I'm trying to implement permissions for tags using a custom field called tenant, so this feature would be very helpful.

**Comment by ypid-work** [comment]
I have not read up on this discussion. I just want to link this issue with #21222 which is a duplicate of this one.
